src/components/MangaControl.tsx

```diff
@@ -55,8 +61,13 @@ const MangaControl: React.FunctionComponent<Props> = (props: Props) => {
   const changePage = (event: React.ChangeEvent<HTMLSelectElement>) => {
     updatePage(+event.currentTarget.value);
   };
+  const changeAutoLoadChapter = () => {
+    updateReaderSettings({ ...readerSettings, autoLoadChapter: !readerSettings.autoLoadChapter })

```

Comment:
seem to be using this function in the form
```
updateReaderSettings({...oldReaderSettings, autoLoadChapter: blah})
```
a lot.

could potentially refactor `updateReaderSettings` to take in a partial reader setting object and destructure it like so:
```
const updateReaderSettings = (newSettings) => (
   newReaderSettings = {
     ...oldReaderSettings,
     ...newSettings
  }
)
```
then you can just use it like
```
updateReaderSettings({autoLoadChapter: blah})
```

Reply:
wait what type is it going to be tho, not too sure how to define newSettings type

Reply:
probably `Nullable<SettingsType>`